Language/Haskell/GhcMod/ImportedFrom.hs

```diff
@@ -0,0 +1,785 @@
+  where
+    -- f symbol export = filter (symbol ==) thisExports
+    f sym export = filter (postfixMatch sym) thisExports
+       where thisExports = qualifiedExports export         -- Things that this module exports.
+
+refineLeadingDot :: MySymbol -> [ModuleExports] -> [ModuleExports]
+refineLeadingDot (MySymbolUserQualified _)           exports = exports
+refineLeadingDot (MySymbolSysQualified symb)         exports = map (\e -> e { qualifiedExports = f leadingDot e }) exports
+  where
+    leadingDot :: String
+    leadingDot = '.' : last (separateBy '.' symb)
+
+    -- f symbol export = filter (symbol ==) thisExports
+    f symbol export = filter (symbol `isSuffixOf`) thisExports
+       where thisExports = qualifiedExports export         -- Things that this module exports.
+
+refineVisibleExports
+    :: forall m. (GhcMonad m, MonadIO m, GmOut m, GmLog m)
+    => (String -> m (Maybe String))
+    -> [ModuleExports]
+    -> m [ModuleExports]
+refineVisibleExports getHaddockInterfaces exports = mapM f exports
+  where
+    f :: ModuleExports -> m ModuleExports
+    f mexports = do
+        let pname          = mPackageName     mexports -- e.g. "base-4.8.2.0"
+            thisModuleName = mName            mexports -- e.g. "Prelude"
+            qexports       = qualifiedExports mexports -- e.g. ["base-4.8.2.0:GHC.Base.Just", ...]
+        visibleExportsMap <- getVisibleExports getHaddockInterfaces pname
+        gmLog GmDebug "" $ strDoc $ "visibleExportsMap: " ++ show visibleExportsMap
+
+        let thisModVisibleExports = fromMaybe
+                                        (error $ "Could not get visible exports of " ++ pname)
+                                        (join $ traverse (M.lookup thisModuleName) visibleExportsMap)
+
+        let qexports' = filter (hasPostfixMatch thisModVisibleExports) qexports
+
+        gmLog GmDebug "" $ strDoc $ show (qexports, qexports')
+
+        return $ mexports { qualifiedExports = qexports' }
+
+    -- hasPostfixMatch "base-4.8.2.0:GHC.Base.Just" ["Just", "True", ...] -> True
+    hasPostfixMatch :: [String] -> String -> Bool
+    hasPostfixMatch xs s = last (separateBy '.' s) `elem` xs
+
+-- | The last thing with a single export must be the match? Iffy.
+getLastMatch :: [ModuleExports] -> Maybe ModuleExports
+getLastMatch exports = Safe.lastMay $ filter f exports
+  where
+    f me = length (qualifiedExports me) == 1
+
+-- | Try to look up the Haddock URL for a symbol.
+guessHaddockUrl
+    :: forall m.
+       (GhcMonad m, MonadIO m, GmOut m, GmLog m)
+    => ModSummary
+    -> FilePath
+    -> String
+    -> String
+    -> Int
+    -> Int
+    -> FilePath
+    -> (FilePath -> [String] -> String -> IO String)
+    -> [GhcPkgDb]
+    -> m (Either String String)
+guessHaddockUrl modSum targetFile targetModule symbol lineNr colNr ghcPkg readProc pkgDbStack = do
+    gmLog GmDebug "" $ strDoc $ "guessHaddockUrl: targetFile: "   ++ targetFile
```

Comment:
Changed these everywhere.

```
commit f52077c9afa32164b253a46df236f0fa4a7c4d0f
Author: Carlo Hamalainen <carlo@carlo-hamalainen.net>
Date:   Sun Jul 17 22:32:20 2016 +1000

    Use parameter of ```gmLog GmDebug``` instead of passing ```""```.
```